interfaces/api/service/IFrontEnd.h

```diff
@@ -32,6 +33,39 @@ namespace SolAR {
 namespace api {
 namespace service {
 
+/**
+ * @typedef DeviceType
+ * @brief <B>Define the types of all devices.</B>
+ */
+typedef enum {
+    OTHER_DEVICE = 0,
+    HOLOLENS2_HEADSET = 1,
+    LYNX_HEADSET = 2,
+    ANDROID_DEVICE = 3,
+    IOS_DEVICE = 4
+} DeviceType;
```

Comment:
I probably miss the big picture here, but how do these values actually impact the code or does it only serve an informational purpose?

If it's only for rendering, we could have values such as: phone, tablet, foldable, AR/MR/VR headset, etc... ?

If we want to identify precisely a platform, this won't probably scale well: is LYNX_HEADSET will be applicable for all future Lynx products? Android device can also mean VR headset, or a lot of other devices, etc...

So according to what it is used for, we could have

```cpp
enum class DeviceType { phone, tablet, headset,... } // can be used to coarse grain filter a list, or render a 3D mesh representing it
```
And a type representing precisely the device

```mermaid
classDiagram
   class DeviceDescription {
         +DeviceType type
         +String brand
         +String modelReference
    }
```

I guess for now it's good enough as is, but how futureproof do you think this approach is?

Reply:
The idea is that the DeviceType represent the type of device that we currently support in SolAR, for the viewer to quickly distinguish them by their enum value without needing to parse the model string.

Reply:
Yes I get this is handy, but I was wondering about how this could evolve with time. You mention the viewer, that's why I suggested general categories of devices for visualization, given that we could augment the device avatar with textual description to better discrimintate between users. We could discuss about this offline, for me to better understand what's at stake here.

Reply:
I suggest not to modifiy these structures for this version, to avoid to many modifications on API, proto (proxy) and clients. But we need to have a discussion on this topic for next version. OK ?

Reply:
Additionnally, this being the framework project i.e. not the actual implementation, I'm not sure it makes sense to have here a list of currently supported devices (if there is no impact on the framework interfaces that is).
Anybody is free to add support for any device, from this project.

Reply:
You're right: we need to propose another solution for next version